File: How_To_Program_Exercises/JavaHowToProgram9/Chapter3/ComputerizationOfHealthRecords/HealthProfileTest.java
//
//	HealthProfile Tester
//

import java.util.Scanner;

class HealthProfileTest
{
	public static void main (String[] args)
	{
		String fName, lName, gender;
		int currDay, currMonth, currYear;
		int birthDay, birthMonth, birthYear;
		int height, weight;
		
		Scanner input = new Scanner (System.in);
		
		//Prompt first and last names
		System.out.print("Enter your first and last names: ");
		fName = input.next();
		lName = input.next();
		
		//Prompt gender
		System.out.print("Enter your gender (Male/Female): ");
		gender = input.next();
		
		//Prompt date of birth 
		System.out.print("Enter your date of birth (d m y): ");
		birthDay = input.nextInt();
		birthMonth = input.nextInt();
		birthYear = input.nextInt();
		
		//Prompt current date
		System.out.print("Enter the current date (d m y): ");
		currDay = input.nextInt();
		currMonth = input.nextInt();
		currYear = input.nextInt();
		
		//Prompt weight and height
		System.out.print("Enter your height in inches: ");
		height = input.nextInt();
		System.out.print("Enter your weight in pounds: ");
		weight = input.nextInt();
		
		HealthProfile profile = new HealthProfile (fName, lName, gender, birthDay, birthMonth, birthYear, height, weight);
		
		
		
		System.out.println("\n\n\nYour information\n-----------------");
		System.out.println("Name: " + profile.getFirstName() + " " + profile.getLastName());
		System.out.printf("\nGender: %s\nAge: %d\n", profile.getGender(), profile.getAge(currDay, currMonth, currYear));
		System.out.printf("\n\nHeight: %din\nWeight: %dlbs", profile.getHeight(), profile.getWeight());
		
		System.out.println("\n\nMaximum Heart Rate: " + profile.getMaximumHeartRate());
		System.out.println("\nTarget Heart Rate: " + profile.getTargetHeartRate());
		
		System.out.println("\nBMI: " + profile.getBMI());
		
		String message = "\n\n\nBMI: BMI VALUES" +
                        "\nUnderweight: less than 18.5" + "\nNormal: between 18.5 and 24.9" +
                        "\nOverweight: between 25 and 29.9" + "\nObese: 30 or greater";

		System.out.println(message);
	} 
}

File: How_To_Program_Exercises/C++How_To_Program_8/Chapter3/HealthProfileClass/HealthProfileTest.cpp
//
//	HealthProfile Class Test
//

#include <iostream>
#include <string>
#include "HealthProfile.h"

int main()
{
	std::string fName, lName, gender;
	int height, weight;
	int birthDay, birthMonth, birthYear;

	//Prompt first name, last name and gender
	std::cout << "Enter your first name and last name: ";
	std::cin >> fName >> lName;

	std::cout << "Enter your gender: ";
	std::cin >> gender;

	std::cout << "Enter your height (in inches) and weight (in pounds): ";
	std::cin >> height >> weight;

	std::cout << "Enter your date of birth in this format d m y: ";
	std::cin >> birthDay >> birthMonth >> birthYear;

	//Create an instance of HealthProfile
	HealthProfile profile(fName, lName, gender, birthDay, birthMonth, birthYear, height, weight);

	//Set the age data member
	profile.setAge();

	std::cout << "\n\nYour information: "
		<< "\n------------------"
		<< "\nName: " << profile.getFirstName()
		<< " " << profile.getLastName()
		<< "\nAge: " << profile.getAge()
		<< "\nGender: " << profile.getGender()
		<< "\nHeight: " << profile.getHeight() << " inches"
		<< "\nWeight: " << profile.getWeight() << " lbs"
		<< "\nMaximum Heart Rate: " << profile.getMaxHeartRate()
		<< "\nTarget Heart Rate: " << profile.getTargetHeartRate()
		<< "\nBMI: " << profile.getBMI();

	std::cout << "\n\nBMI Values"
		<< "\nUnderweight: less than 18.5"
		<< "\nNormal: between 18.5 and 24.9"
		<< "\nOverweight: between 25 and 29.9"
		<< "\nObese: 30 or greater"
		<< std::endl << std::endl;


	
}